### PR TITLE
Allow sc.show with non-dim coords

### LINKED ISSUE #1499: Confusing error when showing DataArray with mismatch in coord dims
The call of `sc.show` in this example raises an `IndexError`.

```.py
import scipp as sc
import numpy as np

N = 10
var = sc.Variable(unit=sc.units.m, dims=['x'], values=np.random.normal(3, 5, N))
dset = sc.DataArray(var, coords={'asd': sc.Variable(dims=['y'], values=np.linspace(0, 1, N))})

sc.show(dset)
```
```
---------------------------------------------------------------------------

IndexError                                Traceback (most recent call last)

<ipython-input-6-5a189de14405> in <module>
      1 
----> 2 sc.show(dset)
      3 

~/Work/scipp-devel/install-release/scipp/show.py in show(container)
    453     """
    454     from IPython.core.display import display, HTML
--> 455     display(HTML(make_svg(container)))

~/Work/scipp-devel/install-release/scipp/show.py in make_svg(container)
    445     else:
    446         draw = DatasetDrawer(container)
--> 447     return draw.make_svg()
    448 
    449 

~/Work/scipp-devel/install-release/scipp/show.py in make_svg(self)
    381                 elif var.dims[-1] == dims[-1]:
    382                     area_x.append(item)
--> 383                 elif var.dims[-1] == dims[-2]:
    384                     area_y.append(item)
    385                 else:

IndexError: list index out of range
```

This happens whenever the `dims` of the coordinates differ from the `dims` of `var`.

### PR DESCRIPTION
**Pull request by jl-wynen**
EDIT
Changed to throw a `DimensionError` instead of just not showing coords in the SVG.

Not sure if this still fixes issue #1499.

### CODE REVIEW

**Comment by nvaytet** [comment]
I think I agree with @SimonHeybrock 's comment here https://github.com/scipp/scipp/issues/1499#issuecomment-757643519
Unless this is actually causing some notebooks to fail, and the quick `try/except` fix would be fine, I would say it's better to talk about whether such a DataArray is a valid thing to have.
When you are extracting a DataArray from a Dataset, it drops all the coords whose dimensions don't depend on the data dimensions. So the `y` coord in your example would never show up in the resulting DataArray.
Of course, you can create it as you did, but do we have a use for this case?
Or should this be forbidden, as Simon suggests.

**Comment by nvaytet** [comment]
In addition, i'm not sure just dropping the coord from the svg repr is the right thing to do, because the user might just think there is no coord there. The html repr correctly lists the coord, so maybe if we do allow such data arrays, the coord should also appear in the svg?

File: python/src/scipp/show.py [hunk 1]
```diff
@@ -380,14 +380,25 @@ def make_svg(self):
...
+                        area_0d.append(item)
+                    elif var.dims[-1] == dims[-1]:
+                        area_x.append(item)
+                    elif var.dims[-1] == dims[-2]:
+                        area_y.append(item)
+                    elif var.dims[-1] == dims[-3]:
+                        area_z.append(item)
+                    else:
+                        # dirty hack to trigger error handling below
+                        raise IndexError()
```
Comment (nvaytet) [inline python/src/scipp/show.py:394]: I'm not sure I understand why you need the `try` block with the `raise IndexError` and not just have
```Py
else:
    raise sc.DimensionError(
                        "Cannot show DataArray, it contains a coordinate with"
                        f" a dimension label ('{var.dims[-1]}') that is not "
                        "contained in the data") from None
```

Comment (jl-wynen) [reply python/src/scipp/show.py:394]: If the data array is e.g. 2 dimensional, the last `elif` will throw. The `try``-except` is just a way of not writing `if len(dims) > N and ...` in every branch.

**Comment by jl-wynen** [comment]
@nvaytet Can this be merged?

**Comment by SimonHeybrock** [comment]
Can you give an example of a data array that triggers this branch?

**Comment by jl-wynen** [comment]
Here #1499.

**Comment by SimonHeybrock** [comment]
This implies that we cannot `show` a slice of data with a bin-edge coord, e.g., `data['tof', 17]`?

**Comment by jl-wynen** [comment]
Yes, because the slice does not have the necessary information to align the boxes with any axis of the data. The only alternative that I see is my original implementation which simply drops all meta data that cannot be displayed.
Also, any non-scalar attribute which has a dimension that is not in the data causes the same error

**Comment by SimonHeybrock** [comment]
Suggest to close in favor of #1784.

**Comment by nvaytet** [comment]
@jl-wynen is it ok to close this in favor of @SimonHeybrock 's PR?

**Comment by SimonHeybrock** [comment]
This is probably even less relevant with the ownership rewrite, since we do not support coords exceeding data araay dims any more (except for bin edges).

**Comment by jl-wynen** [comment]
Oh right. I thought this was already closed

**Pull request closed without merging**